### PR TITLE
Resurrecting GraphQL PR

### PR DESCRIPTION
**Pull request by jnwng**
This is a revival of [#1642](https://github.com/prettier/prettier/pull/1642) with modifications after rebasing.

### CODE REVIEW

**Comment by jnwng** [comment]
@stubailo all the tests pass on this branch now. does it make sense to open this against prettier:master and just keep adding tests until we're satisified? or should we limit the noise upstream and add tests here?

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "flow-parser": "0.47.0",
     "get-stdin": "5.0.1",
     "glob": "7.1.2",
+    "graphql": "^0.10.1",
```
Comment (vjeux) [inline package.json]: Please use exact one

File: docs/index.js [hunk 2]
```diff
@@ -8370,155 +8370,155 @@ index$4.hasColor = hasColor;
 index$4.stripColor = stripColor;
 index$4.supportsColor = supportsColor_1;
 
-var index$26 = {
```
Comment (vjeux) [inline docs/index.js]: Can you revert the docs changes, I only ship new ones when I do a release.

File: scripts/build/build.sh [hunk 3]
```diff
@@ -45,6 +45,9 @@ node_modules/.bin/rollup -c scripts/build/rollup.docs.config.js --environment fi
 echo 'Bundling docs flow...';
 node_modules/.bin/rollup -c scripts/build/rollup.docs.config.js --environment filepath:src/parser-flow.js
 
+echo 'Bundling docs graphql...';
```
Comment (vjeux) [inline scripts/build/build.sh:51]: You're missing the one for release as well

File: src/parser-graphql.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+"use strict";
+
+function parse(text) {
+  // Inline the require to avoid loading all the JS if we don't use it
+  const parser = require("graphql/language");
+  const ast = parser.parse(text);
+  return ast;
```
Comment (vjeux) [inline src/parser-graphql.js]: You'll probably want to use create-error in order to get nice looking errors

File: src/printer-graphql.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    case "Argument": {
+      return concat([
+        path.call(print, "name"),
+        ": ",
+        path.call(print, "value")
+      ]);
+    }
+
+    default:
+      throw new Error("unknown graphql type: " + JSON.stringify(n.type));
```
Comment (vjeux) [inline src/printer-graphql.js]: Should be n.kind for graphql

**Review by vjeux — COMMENTED** [review]
Awesome! Feel free to send a PR with this to prettier and i'll merge it, this way we can have multiple people iterating on the same time.

**Comment by jnwng** [comment]
merged as part of https://github.com/prettier/prettier/pull/1982

**Pull request closed without merging**